zjit/src/hir.rs

```diff
@@ -1334,6 +1344,14 @@ impl Function {
                         let replacement = self.push_insn(block, Insn::Const { val: Const::Value(unsafe { (*ice).value }) });
                         self.make_equal_to(insn_id, replacement);
                     }
+                    Insn::ObjToString { val, call_info, cd, state, .. } => {
+                        if self.is_a(val, types::StringExact) {
+                            self.make_equal_to(insn_id, val);
+                        } else {
+                            let replacement = self.push_insn(block, Insn::SendWithoutBlock { self_val: val, call_info, cd, args: vec![], state });
+                            self.make_equal_to(insn_id, replacement)
+                        }
+                    }
```

Comment:
Maybe better for a follow-up PR since this PR looks pretty done, but perhaps we should handle this instead by generating a send and then having a rewrite pass that rewrites all `a.to_s` when `a` is a `StringExact` into `a`. That is a little more generic

Reply:
I believe @ywenc started with that approach, but it didn't work so well because in the general case of that optimization we'd want a method redefined patch point, but for `objtostring` specifically we don't want the patch point.

Reply:
Can you leave a comment about not having a patchpoint in this rewrite? Seems like a potential future point of confusion